runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaCacheClientFetchFactory.java

```diff
@@ -1102,6 +1107,15 @@ private void doClientReplyDataIfNecessary(
             {
                 final KafkaCacheEntryFW nextEntry = cursor.next(entryRO);
 
+                if (nextEntry == null &&
+                        group.partitionOffset >= targetOffset.getAsLong() &&
+                        !group.dataFrameReceived &&
+                        cursor.offset == 0)
```

Comment:
`cursor.offset` might never be `0` if Kafka has cleaned up old messages, then earliest offset in the partition would be `> 0`.